### PR TITLE
[V1.2] Added moduleName and AddModuleCommandParser

### PR DESCRIPTION
**Pull request by kester-ng**
Added the above plus the JUNIT test cases
Only the parsers and commands have no JUNIT test cases for now. Require the model to be able to start (which is more or less done)

### CODE REVIEW

File: src/main/java/NASA/model/module/ModuleName.java [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+
+import static java.util.Objects.requireNonNull;
+import static NASA.commons.util.AppUtil.checkArgument;
+
+public class ModuleName {
+
+    public static final String MESSAGE_CONSTRAINTS =
+            "Module name should only contain alphanumeric characters and spaces, and it should not blank";
+
+    public static final String VALIDATION_REGEX = "[\\p{Alnum}][\\p{Alnum} ]*";
```
Comment (EdmondOng) [inline src/main/java/NASA/model/module/ModuleName.java]: Maybe ModuleName regex should disallow numbers?

File: src/test/java/NASA/logic/parser/ParserUtilTest.java [hunk 2]
```diff
@@ -21,11 +22,13 @@
     private static final String VALID_NOTE = "Finish milestone v1.2 by next wednesday."
             + "prepare for new features.";
     private static final String VALID_PRIORITY = "3";
+    private static final String VALID_MODULE_NAME = "CS2100";
```
Comment (EdmondOng) [inline src/test/java/NASA/logic/parser/ParserUtilTest.java]: What is the difference between ModuleName and ModuleCode? My understanding is moduleCode can be "CS2103" while name is more general like "Software Engineering".

Comment (kester-ng) [reply src/test/java/NASA/logic/parser/ParserUtilTest.java]: Oops i supposed to put computer organisation, i changed it alr

File: src/main/java/NASA/model/module/ModuleName.java (hunk 1 above)
Comment (kester-ng) [reply src/main/java/NASA/model/module/ModuleName.java]: Module names may have numbers! Like people can call programming methodology 2 instead of II.

**Pull request merged**